### PR TITLE
feat(query): add last_day, previous_day, next_day function

### PR DESCRIPTION
**Pull request by Freejww**
I hereby agree to the terms of the CLA available at: https://docs.databend.com/dev/policies/cla/

## Summary
Support function: last_day, next_day and previous_day.

fixes: #16715 

## Tests

- [x] Unit Test
- [x] Logic Test
- [ ] Benchmark Test
- [ ] No Test - _Explain why_

## Type of change

- [ ] Bug Fix (non-breaking change which fixes an issue)
- [x] New Feature (non-breaking change which adds functionality)
- [ ] Breaking Change (fix or feature that could cause existing functionality not to work as expected)
- [ ] Documentation Update
- [ ] Refactoring
- [ ] Performance Improvement
- [ ] Other (please describe):

- - -
This change is [Reviewable](https://reviewable.io/reviews/databendlabs/databend/16740)

### CODE REVIEW

**Comment by Freejww** [comment]
Hi, I ran failed ci tests locally, the result is 0. PTAL.  @TCeason

**Comment by TCeason** [comment]
> Hi, I ran failed ci tests locally, the result is 0. PTAL.  @TCeason 

https://github.com/databendlabs/databend/pull/16739 fix it.

**Comment by Freejww** [comment]
> > Hi, I ran failed ci tests locally, the result is 0. PTAL.  @TCeason
> 
> #16739 fix it.

Thanks~

**Comment by Freejww** [comment]
Hi，this pr is ready for review. @TCeason

File: tests/sqllogictests/suites/query/functions/02_0012_function_datetimes.test [hunk 1]
```diff
@@ -1170,6 +1170,116 @@ select to_datetime('2022-04-01 06:50:20')   < '2022-04-02 04:50:20'
...
+----
+2024-10-28
+
+query T
+select next_day(to_timestamp('2024-10-25 09:38:18.165575'), friday);
+----
+2024-11-01
+
+query T
+select next_day(to_timestamp('2024-10-25 09:38:18.165575'), sunday);
```
Comment (TCeason) [inline tests/sqllogictests/suites/query/functions/02_0012_function_datetimes.test:1279]: Fine. And better add diff timezone test.

Like set timezone='Asia/Shanghai'; then test timestamp `2024-10-25 00:38:18.165575` and `2024-10-25 23:38:18.165575`

Comment (Freejww) [reply tests/sqllogictests/suites/query/functions/02_0012_function_datetimes.test:1279]: Thanks~, have done.

Comment (TCeason) [reply tests/sqllogictests/suites/query/functions/02_0012_function_datetimes.test:1279]: > Thanks~, have done.

Good test. And time part should be Like `2024-10-25 00:38:18.165575 and 2024-10-25 23:38:18.165575`.

We need to test a scenario where the date changes due to a time zone change.

e.g. `2024-10-25 01:00:00 ` at Asia/Shanghai is `2024-10-25 01:00:00`

at UTC  is `2024-10-24 17:00:00`

**Review by TCeason — APPROVED** [review]

**Comment by wubx** [comment]
cc @soyeric128  need doc

**Comment by BohuTANG** [comment]
Hi @soyeric128, please update the doc. Thank you!

**Pull request merged**